crates/ursa-gateway/src/server/route/api/v1/get.rs

```diff
@@ -76,6 +79,37 @@ pub async fn get_block_handler(
 
     debug!("received indexer response for {cid}:\n{indexer_response:?}");
 
+    let resp = match get_provider(&indexer_response) {
+        Some(addrs) => {
+            let mut addr_iter = addrs.iter();
+
+            loop {
+                match addr_iter.next() {
+                    Some(&addr) => {
+                        let server_config = ServerConfig::new(4069, addr.to_string());
+                        let params = NetworkGetParams { cid: cid.clone() };
+                        match ursa_rpc_client::functions::get_block(params, server_config).await {
+                            Ok(resp) => break resp,
+                            Err(_) => error!("request to RPC server failed"),
```

Comment:
Possibility in the case of 404 that the cache node doesn't keep the content.

Reply:
Passing 404 response now